vm/src/stdlib/os.rs

```diff
@@ -210,6 +210,16 @@ pub(crate) fn fspath(
 
 impl TryFromObject for PyPathLike {
     fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        let obj = match_class!(match obj {
+            a @ PyByteArray => {
+                a.borrow_buf_mut().clone().into_pyobject(vm)
+            }
+            m @ PyMemoryView => {
+                let b: PyBytes = m.try_bytes(vm, |x| Vec::from(x))?.into();
+                b.into_pyobject(vm)
+            }
```

Comment:
Rather than checking PyMemoryView directly, CPython is checking if it is a buffer or not. Try to getting buffer from PyObjectRef is like this:

```rust
let buffer = PyBufferRef::TryFromObject(obj)
```

Reply:
Changed as suggested!